File: 0070-climbing-stairs/0070-climbing-stairs.py
class Solution(object):
    def climbStairs(self, n):
        if n == 2 or n ==1:
            return n
        a = 1
        b = 2
        for i in range(3,n+1):
            c = a+b
            a = b
            b = c
        return b        

File: 0070-climbing-stairs/0070-climbing-stairs.java
class Solution {
    static int[] dp;
    public int climbStairs(int n) {
        dp = new int[n+2];
        Arrays.fill(dp,-1);
        return stairs(n+1);

    }
    public int stairs(int n) {
        if(n == 0 || n== 1) return n;
        if(dp[n] != -1) return dp[n];
        int ans = stairs(n-1)+stairs(n-2); 
        return dp[n] = ans;
    }
}